projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6441,6 +6526,26 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         }
     }
 
+    public endAdd(commit = true, event?: Event) {
+        const row = this.crudService.row;
+        const cell = this.crudService.cell;
+        if (!row && !cell) {
+            return;
+        }
+        if (commit) {
+            this.gridAPI.submit_add_value();
```

Comment:
This seems to generate an update transaction. There's no need to generate an update transaction before adding the row.